app/react/src/client/preview/types-6-0.ts

```diff
@@ -1,10 +1,10 @@
-import { Component, FunctionComponent } from 'react';
+import { ComponentClass, FunctionComponent } from 'react';
 import { Args as DefaultArgs, Annotations, BaseMeta, BaseStory } from '@storybook/addons';
 import { StoryFnReactReturnType } from './types';
 
 export { Args, ArgTypes, Parameters, StoryContext } from '@storybook/addons';
 
-type ReactComponent = Component | FunctionComponent<any>;
+type ReactComponent = ComponentClass<any> | FunctionComponent<any>;
```

Comment:
There's a `React.ComponentType<Props>` type that can replace this perhaps?

Reply:
Thanks @disintegrator https://github.com/storybookjs/storybook/pull/12110